languages/ruby/lib/oso/polar/polar.rb

```diff
@@ -28,11 +31,25 @@ def clear
       # @param name [String]
       # @raise [PolarFileExtensionError] if provided filename has invalid extension.
       # @raise [PolarFileNotFoundError] if provided filename does not exist.
-      def load_file(name)
+      def load_file(name) # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
         raise PolarFileExtensionError unless ['.pol', '.polar'].include? File.extname(name)
         raise PolarFileNotFoundError, name unless File.file?(name)
 
-        load_queue << name
+        hash = hash_file(name)
+
+        if @loaded_names.key?(name)
+          raise PolarRuntimeError, "File #{name} has already been loaded." unless @loaded_names[name] != hash
```

Comment:
bit easier to understand than the double negative, imo

```suggestion
          raise PolarRuntimeError, "File #{name} has already been loaded." if @loaded_names[name] == hash
```

Reply:
yeah, was like this but got a rubocop error

Reply:
I can put the guard clause on the other error